test/unit/adapters/http.js

```diff
@@ -179,6 +179,41 @@ describe('supports http with nodejs', function () {
     });
   });
 
+  it('should support transparent brotli', function (done) {
+    var data = {
+      firstName: 'Fred',
+      lastName: 'Flintstone',
+      emailAddr: 'fred@example.com'
+    };
+
+    // for old node version
+    if (typeof zlib.brotliCompress !== 'function'){
```

Comment:
I don't think this is a good way to have test coverage, since the coverage will depend on the version of the environment and always run just 1 test instead of both, can you try adding separate tests for each condition? maybe by creating a context where zlib doesn't have this function

Reply:
also should this check for `createBrotliDecompress`? or is `brotliCompress` fine? not sure if I'm missing something here?

Reply:
> also should this check for `createBrotliDecompress`? or is `brotliCompress` fine? not sure if I'm missing something here?

All brotli function came into node same time. If there is one, the others must be.

https://nodejs.org/api/zlib.html#zlib_class_brotlioptions
https://nodejs.org/api/zlib.html#zlib_class_zlib_brotlicompress
https://nodejs.org/api/zlib.html#zlib_class_zlib_brotlidecompress
https://nodejs.org/api/zlib.html#zlib_zlib_brotlicompress_buffer_options_callback
https://nodejs.org/api/zlib.html#zlib_zlib_brotlicompresssync_buffer_options
https://nodejs.org/api/zlib.html#zlib_zlib_brotlidecompress_buffer_options_callback
https://nodejs.org/api/zlib.html#zlib_zlib_brotlidecompresssync_buffer_options

Reply:
> I don't think this is a good way to have test coverage, since the coverage will depend on the version of the environment and always run just 1 test instead of both, can you try adding separate tests for each condition? maybe by creating a context where zlib doesn't have this function

Sorry I cant, it is enough for me.